company-ycmd.el

```diff
@@ -95,6 +95,12 @@ feature."
   :type 'boolean
   :group 'company-ycmd)
 
+(defcustom company-ycmd-request-sync-timeout 0.05
```

Comment:
My guess is that most of our users are using ycmd through company. If the synchronous behavior improves the company experience (e.g. by reducing flickering) then I think synchronous is the better default.